Cargo.toml

```diff
@@ -12,6 +12,19 @@ license = "Apache-2.0 OR MIT"
 libc = ">=0.2.39"
 kvm-bindings = { version = ">=0.2.0", features = ["fam-wrappers"] }
 vmm-sys-util = ">=0.2.1"
+bitflags = { version = ">=1.2.1", optional = true }
+sev = { path = "./sev", features = ["openssl"], optional = true }
```

Comment:
There's no `sev` crate, does this refer to the `sev` mod?

Reply:
~Or did you have the `sev` crate locally in your workspace? The PR doesn't build without it.~
LE: I *think* this is the `sev` crate from the [enarx](https://github.com/enarx/enarx/tree/e044d49cb8c0e19bad8cf98b42129ff89bed1d8e/sev) repo; but, as `enarx` doesn't have a `Cargo.toml` (they seem to use a different build system, `cargo make`?), Cargo can't get the crate dependency right with
```toml
sev = { git = "https://github.com/enarx/enarx.git", features = ["openssl"], optional = true }
```
This would be a blocker in publishing a new version of `kvm-ioctls` with `sev` support: the `sev` crate needs to be published first.
In the meantime, for playing around, maybe you could get away with a multiple-dependency like explained [here](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations). Cargo would look for the `sev` crate on crates.io and, until it finds it there, pick it up from the local repo (to get the CI to pass, you'd have to yank it into your fork and add it to the PR).

Reply:
What's more, maybe you should also restrict the `sev` dependencies to `target=x86_64` as they won't build on ARM.

Reply:
> ~Or did you have the `sev` crate locally in your workspace? The PR doesn't build without it.~
> LE: I _think_ this is the `sev` crate from the [enarx](https://github.com/enarx/enarx/tree/e044d49cb8c0e19bad8cf98b42129ff89bed1d8e/sev) repo; but, as `enarx` doesn't have a `Cargo.toml` (they seem to use a different build system, `cargo make`?), Cargo can't get the crate dependency right with
> 
> ```toml
> sev = { git = "https://github.com/enarx/enarx.git", features = ["openssl"], optional = true }
> ```
> 
> This would be a blocker in publishing a new version of `kvm-ioctls` with `sev` support: the `sev` crate needs to be published first.
> In the meantime, for playing around, maybe you could get away with a multiple-dependency like explained [here](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations). Cargo would look for the `sev` crate on crates.io and, until it finds it there, pick it up from the local repo (to get the CI to pass, you'd have to yank it into your fork and add it to the PR).

Yes, I do have a modified `sev` crate in my workspace, which also implements the traits to resolve compilation issues you mentioned below. (that is why I marked this PR as a draft version ;))
As enarx team has cited in another thread. Let's wait a moment for their coming release of `sev` crate and APIs. I'll have an update based on that.

Reply:
> What's more, maybe you should also restrict the `sev` dependencies to `target=x86_64` as they won't build on ARM.

Yes agree, will update.